addons/crm/models/crm_lead.py

```diff
@@ -0,0 +1,1178 @@
+                    partner_info['full_name'] = '%s <%s>' % (self.partner_name or self.contact_name, email)
+                    break
+        return res
+
+    @api.model
+    def get_empty_list_help(self, help):
+        ctx = dict(self.env.context)
+        ctx['empty_list_help_model'] = 'crm.team'
+        ctx['empty_list_help_id'] = ctx.get('default_team_id', None)
+        ctx['empty_list_help_document_name'] = _("opportunities")
+        if help:
+            alias = self.env.ref("crm.mail_alias_lead_info")
+            if alias and alias.alias_domain and alias.alias_name:
+                dynamic_help = '<p>%s</p>' % _("""All email incoming to %(link)s  will automatically create new opportunity.
+Update your business card, phone book, social media,... Send an email right now and see it here.""") % {
+                    'link': "<a href='mailto:%s'>%s</a>" % (alias.alias_name, alias.alias_domain)
+                }
+                return '<p class="oe_view_nocontent_create">%s</p>%s%s' % (
+                    _('Click to add a new opportunity'),
+                    help,
+                    dynamic_help)
+        return super(CrmLead, self.with_context(ctx)).get_empty_list_help(help)
+
+    @api.multi
+    def stage_find(self, team_id, domain=None, order='sequence'):
+        """ Override of the base.stage method
+            Parameter of the stage search taken from the lead:
+            - type: stage type must be the same or 'both'
+            - team_id: if set, stages must belong to this team or
+              be a default stage; if not set, stages must be default
+              stages
+        """
+        # check whether we should try to add a condition on type
+        avoid_add_type_term = any([term for term in domain if len(term) == 3 if term[0] == 'stage_type'])
+        # collect all team_ids
+        team_ids = set()
+        types = ['both']
+        default_type = self.env.context.get('default_type')
+        if default_type:
+            ctx_type = default_type
+            types += [ctx_type]
+        if team_id:
+            team_ids.add(team_id)
+        for lead in self:
+            if lead.team_id:
+                team_ids.add(lead.team_id.id)
+            if lead.lead_type not in types:
+                types.append(lead.lead_type)
+        # OR all team_ids
+        search_domain = []
+        if team_ids:
+            search_domain += [('|')] * (len(team_ids) - 1)
+            for team_id in team_ids:
+                search_domain.append(('team_ids', '=', team_id))
+        # AND with cases types
+        if not avoid_add_type_term:
+            search_domain.append(('stage_type', 'in', types))
+        # AND with the domain in parameter
+        search_domain += list(domain)
+        # perform search, return the first found
+        stage = self.env['crm.stage'].search(search_domain, order=order, limit=1)
+        return stage.id
+
+    def get_duplicated_leads(self, partner_id, include_lost=False):
```

Comment:
from where this method is called ? 
method name is changed and directly defined in wizard. 
_get_duplicated_leads.